### PR TITLE
Feature/add project

### PR DESCRIPTION
**Pull request by Jaranag**
· Added feature to add new projects to the projects list
· Added typing in types folder: Also added project typing.
· Added project form to add new projects with validation
· Added uuidv4 to set ids automatically
· Added default styles. Subject to changes

### CODE REVIEW

**Review by JiajunQiuWu — APPROVED** [review]
esta muy bien

**Review by rekloyd — APPROVED** [review]

**Review by JoseAlfredoMiguel — APPROVED** [review]

File: src/components/Sidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import { useState } from 'react'
+import { Project } from '../types/index.tsx'
+import { ProjectForm } from './ProjectForm'
+import { v4 as uuidv4 } from "uuid";
+
+uuidv4();
```
Comment (SaraVictorFernandez) [inline src/components/Sidebar.tsx:7]: qué hace este uuidv4?

File: src/types/index.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export interface Project {
+  id: string;
+  projectName: string;
+  completed: boolean;
+  isEditing: boolean;
+}
+
+export interface ProjectFormProps {
```
Comment (SaraVictorFernandez) [inline src/types/index.tsx:8]: Component Props are usually declared on the Component's file to improve readability

File: src/components/Sidebar.tsx (hunk 1 above)
Comment (Jaranag) [reply src/components/Sidebar.tsx:7]: Es el recurso que usamos para añadir un id al objeto de proyecto. Ejemplo: '11bf5b37-e0b8-42e0-8dcf-dc8c4aefc000'

File: src/types/index.tsx (hunk 2 above)
Comment (Jaranag) [reply src/types/index.tsx]: Prop declaration now in same file as component

**Review by SaraVictorFernandez — APPROVED** [review]
looks good to me

**Pull request merged**